Automattic/wp-calypso
# Headers: Remove usage of old ThemesHeader component

Follow-up of https://github.com/Automattic/wp-calypso/pull/83087
PT: p9Jlb4-8ec-p2

## Proposed Changes

* Removed old `ThemesHeader` component usage

## Testing Instructions



* Ensure the logged-in /themes work as production: http://calypso.localhost:3000/themes
* Ensure the logged-out /themes work as production: http://calypso.localhost:3000/themes
* Ensure the http://calypso.localhost:3000/themes/upload/{site} work as production (from `Install new theme` CTA)

## Pre-merge Checklist
